### PR TITLE
Fix user lockout migrations

### PR DESCRIPTION
**Pull request by deanmarcussen**
Fixes https://github.com/OrchardCMS/OrchardCore/issues/9641
Fixes https://github.com/OrchardCMS/OrchardCore/issues/9654
Fixes https://github.com/OrchardCMS/OrchardCore/issues/9658

Just needs running on MySql to see if it actually fixes it

### CODE REVIEW

**Comment by deanmarcussen** [comment]
Passed https://github.com/OrchardCMS/OrchardCore/actions/runs/935949713

File: src/OrchardCore.Modules/OrchardCore.Users/Migrations.cs [hunk 1]
```diff
@@ -123,14 +126,21 @@ public int UpdateFrom4()
...
-            }
+                var session = scope.ServiceProvider.GetRequiredService<ISession>();
+                var users = await session.Query<User>().ListAsync();
+                foreach (var user in users)
+                {
+                    user.UserId = user.UserName;
+                    session.Save(user);
+                }
+
+                await session.SaveChangesAsync();
```
Comment (jtkech) [inline src/OrchardCore.Modules/OrchardCore.Users/Migrations.cs:142]: Each deferred task is executed in its own scope, so normally `SaveChangesAsync()` is automatically called before disposing the scope. Can you retry it but without this line.

File: src/OrchardCore.Modules/OrchardCore.Users/Migrations.cs [hunk 2]
```diff
@@ -144,15 +154,25 @@ public int UpdateFrom6()
...
+                {
+                    user.UserName = user.UserName.Replace('@', '+');
+                    user.NormalizedUserName = user.NormalizedUserName.Replace('@', '+');
+                    session.Save(user);
+                }
+
+                if (users.Any())
+                {
+                    await session.SaveChangesAsync();
+                }
```
Comment (jtkech) [inline src/OrchardCore.Modules/OrchardCore.Users/Migrations.cs:174]: Idem as above

**Review by jtkech — COMMENTED** [review]
@deanmarcussen 

LGTM, the only thing is that you don't need to explicitly call `SaveChangesAsync()`.

I let you retry it without these calls.

File: src/OrchardCore.Modules/OrchardCore.Users/Migrations.cs (hunk 1 above)
Comment (deanmarcussen) [reply src/OrchardCore.Modules/OrchardCore.Users/Migrations.cs]: Cool, I couldn't remember. Tested without, works fine

**Comment by davidhayden** [comment]
Looking forward to this! Thank you for all your effort @deanmarcussen @jtkech 👍

**Review by jtkech — APPROVED** [review]

**Pull request merged**